### ISSUE TITLE
`compatibilityDate` codegen should not update code style

### ISSUE THREAD

**Issue by Aareksio** [nuxt/cli#447]
### Environment

- Operating System: Linux
- Node Version:     v22.1.0
- Nuxt Version:     3.12.3
- CLI Version:      3.12.0
- Nitro Version:    2.9.7
- Package Manager:  pnpm@9.4.0
- Builder:          -
- User Config:      -
- Runtime Modules:  -
- Build Modules:    -

### Reproduction

Run a project without `compabilityDate`, accept suggested change when running `nuxi dev`

### Describe the bug

```ts
export default defineNuxtConfig({
	prop1: {},
	prop2: {},
})
```
becomes
```ts
export default defineNuxtConfig({
	prop1: {},

	prop2: {},

	compabilityDate: '2024-07-03',
})
```

Also removes the newline at the end of the file.

### Additional context

As a developer, I agreed to one change:
> Do you want to update your nuxt.config to set compatibilityDate: '2024-07-03'?

Instead the diff has `n` lines of unrelated change, exploding PR size.

### Logs

```shell-script
ℹ Nuxt now supports pinning the behavior of provider and deployment presets with a compatibility date. We recommend you specify a compatibilityDate in your nuxt.config file, or set an environment variable, such as COMPATIBILITY_DATE=2024-07-03.

✔ Do you want to update your nuxt.config to set compatibilityDate: '2024-07-03'?
Yes

✔ Compatibility date set to 2024-07-03 in nuxt.config.ts     
                                                                                                       
ℹ nuxt.config.ts updated. Restarting Nuxt...
```

**Comment by Aareksio** [comment]
@manniL I do not think this has anything to do with nuxi. The logic for this is in nuxt/nuxt repo:
https://github.com/nuxt/nuxt/blob/48e96a95da4957b9ea280da448fe1f95bb0b41ab/packages/nuxt/src/core/nuxt.ts#L109-L150

And the utility which actually breaks the style is from `c12/update`.
Shall we move to [c12](https://github.com/unjs/c12)?

**Comment by manniL** [comment]
Whoops, you are right.

Moving this back to nuxt. Yes. Could you raise this upstream in the c12 repo? ☺️
